packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol

```diff
@@ -421,86 +456,51 @@ contract BridgeFacet is BaseConnextFacet {
+    (uint32 _domain, bytes32 _id) = s.tokenRegistry.getTokenId(_asset);
 
     // format token id
-    bytes29 tokenId = ConnextMessage.formatTokenId(domain, id);
+    bytes29 tokenId = ConnextMessage.formatTokenId(_domain, _id);
 
     // send message
     return ConnextMessage.formatMessage(tokenId, action);
   }
 
   /**
    * @notice Called via `handle` to manage funds associated with a transaction
-   * @dev Will either (a) credit router or (b) make funds available for execution. Don't
+   * @dev Will either (a) credit router and/or repay Portal or (b) make funds available for execution. Don't
    * include execution here
    */
   function _reconcile(uint32 _origin, bytes memory _message) internal {
-    // parse tokenId and action from message
-    bytes29 msg_ = _message.ref(0).mustBeMessage();
-    bytes29 tokenId = msg_.tokenId();
-    bytes29 action = msg_.action();
+    // Parse and process message
+    (uint256 amount, address local, bytes32 transferId, bytes32 canonical) = _reconcileProcessMessage(_message);
 
-    // assert the action is valid
-    if (!action.isTransfer()) {
-      revert BridgeFacet__reconcile_invalidAction();
-    }
-
-    // load the transferId
-    bytes32 transferId = action.transferId();
-
-    // ensure the transaction has not been handled
-    if (s.reconciledTransfers[transferId]) {
-      revert BridgeFacet__reconcile_alreadyReconciled();
-    }
-
-    // get the token contract for the given tokenId on this chain
-    // (if the token is of remote origin and there is
-    // no existing representation token contract, the TokenRegistry will
-    // deploy a new one)
-    address token = s.tokenRegistry.ensureLocalToken(tokenId.domain(), tokenId.id());
-
-    // load amount once
-    uint256 amount = action.amnt();
+    // get the routers
+    address[] storage routers = s.routedTransfers[transferId];
 
-    // NOTE: tokenId + amount must be in plaintext in message so funds can
-    // *only* be minted by `handle`. They are still used in the generation of
-    // the transferId so routers must provide them correctly to be reimbursed
+    // If fast transfer was made using Portal liquidity, we need to repay
+    uint256 portalTransferAmount = s.aavePortalsTransfers[transferId];
 
-    // TODO: do we need to keep this
-    bytes32 details = action.detailsHash();
-
-    // if the token is of remote origin, mint the tokens. will either
-    // - be credited to router (fast liquidity)
-    // - be reserved for execution (slow liquidity)
-    if (!s.tokenRegistry.isLocalOrigin(token)) {
-      IBridgeToken(token).mint(address(this), amount);
-      // Tell the token what its detailsHash is
-      IBridgeToken(token).setDetailsHash(details);
+    uint256 routersAmount = amount;
+    uint256 pathLen = routers.length;
+    if (portalTransferAmount != 0) {
+      // ensure a router took on credit risk
+      if (pathLen != 1) revert BridgeFacet__reconcile_noPortalRouter();
```

Comment:
what if it's multiple routers? shouldn't this be reverting if pathLength == 0?

Reply:
hm? you only want to allow one router to take on the credit risk. since they're not supplying the liquidity you dont really need to multipath it.